cpp/src/arrow/type.cc

```diff
@@ -1216,6 +1321,18 @@ Result<std::shared_ptr<ArrayData>> FieldPath::Get(const ArrayData& data) const {
   return FieldPathGetImpl::Get(this, data.child_data);
 }
 
+Result<std::shared_ptr<ChunkedArray>> FieldPath::Get(
+    const ChunkedArray& chunked_array) const {
+  if (chunked_array.num_chunks() < 1) {
+    return Status::Invalid("Chunked array must have at least one chunk");
```

Comment:
I was under the impression that it would be considered invalid since `ChunkedArray::MakeEmpty` always allocates one chunk. https://github.com/dinimar/arrow/blob/ec6caeff8477af08679de7a7f12b90d4ba302457/cpp/src/arrow/chunked_array.h#L95-L96

Whether that's reasonable or not, I'm open to just returning an empty `ChunkedArray` here (based on its `type` or `storage_type`) if that makes more sense.

Reply:
Ah, ok.  I very rarely work with chunked arrays and so you may be right.  It is possible to create chunks arrays with 0 chunks though:

```
>>> pa.chunked_array([], pa.int8()).num_chunks
0
```

My comment was mostly for understanding.  I thought maybe a chunk was needed for some kind of inference.

I think I would slightly prefer returning an empty chunked array (with 1 chunk for consistency with `ChunkedArray::MakeEmpty`).

Reply:
@benibus please resolve this converstion. At the moment, it's the last thing to be done before second review from Weston Pace